test/parallel/test-cluster-master-kill.js

```diff
@@ -21,10 +21,17 @@ if (cluster.isWorker) {
   // terminate the cluster process
   worker.once('listening', function() {
     setTimeout(function() {
-      process.exit(0);
+/* Cluster.disconnect() will exercise a different 'graceful' shutdown path.
+   From the perspective of the worker, closing the channel is equivalent
+   to the parent calling process.exit(0). */
+      worker.process._channel.close();
```

Comment:
@sam-github That would be another way to do this - I'm not a huge fan of polling but it may be the lesser evil. I can do another version with your suggestion. Thanks.